test/modellib/ModelLib.hpp

```diff
@@ -0,0 +1,59 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//====-- ModelLib.hpp - Building Models for numerical and benchmark tests -===//
+//
+// Copyright 2022-2022 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file contains the declarations for all the models that can be built.
+// The result of each function is a .so built using the modelName.
+//
+//===----------------------------------------------------------------------===//
+
+#include <string>
+
+#include "llvm/ADT/SmallVector.h"
+
+#include "src/Compiler/CompilerUtils.hpp"
+
+// Padding schemes
+#define AUTO_PAD_NOTSET 0
+#define AUTO_PAD_VALID 1
+#define AUTO_PAD_LOWER 2
+#define AUTO_PAD_UPPER 3
+#define AUTO_PAD_UB 4
+const std::string getAutoPadName(const int autoPad);
+
+// Conv2D
+bool genConv2DModelAndCompile(
```

Comment:
A couple of suggestions:
  - to avoid 'polluting' the global namespace I suggest to use a C++ namespace 
  - consider if it make sense to place these functions in a C++ class

Reply:
So you would like to add a name space for the lib? I though if you don't need this (and almost no-one needs this), they just don't include the header/lib in their code.

A class does not make sense, but if you still think that a name space is needed, then what names do you think would work?

Reply:
Understand, I am fine without namespace.